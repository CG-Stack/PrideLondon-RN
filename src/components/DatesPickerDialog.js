// @flow
import React from "react";
import { TouchableOpacity } from "react-native";
import DatesPicker from "./DatesPicker";
import Dialog from "./Dialog";
import Text from "./Text";
import type { DateOrDateRange } from "../data/date-range";
import { formatDateRange } from "../data/formatters";

type Props = {
  onDatesSelected: (?DateOrDateRange) => void,
  onCancel: () => void,
  visible: boolean
};

type State = {
  dateRange: ?DateOrDateRange
};

class DatesPickerDialog extends React.PureComponent<Props, State> {
  state = {
    dateRange: null
  };

  onDatesChange = (dateRange: DateOrDateRange) => {
    this.setState({ dateRange });
  };

  clear = () => {
    this.setState({ dateRange: null });
  };

  apply = () => {
    this.props.onDatesSelected(this.state.dateRange);
  };

  render() {
    const { dateRange } = this.state;
    const title = dateRange
      ? formatDateRange(dateRange, { dateSuffix: " -" })
      : "Select dates";

    return (
      <Dialog
        applyButtonText="Show XX events"
        headerRight={
          <TouchableOpacity onPress={this.clear}>
            <Text>Clear</Text>
          </TouchableOpacity>
        }
        onApply={this.apply}
        onCancel={this.props.onCancel}
        title={title}
        visible={this.props.visible}
      >
        <DatesPicker onChange={this.onDatesChange} dateRange={dateRange} />
      </Dialog>
    );
  }
}

export default DatesPickerDialog;
