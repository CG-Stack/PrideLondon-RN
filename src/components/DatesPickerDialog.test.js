// @flow
import React from "react";
import { shallow } from "enzyme";
import DatesPickerDialog from "./DatesPickerDialog";

it("renders correctly", () => {
  const output = shallow(
    <DatesPickerDialog onDatesSelected={() => {}} onCancel={() => {}} visible />
  );
  expect(output).toMatchSnapshot();
});
