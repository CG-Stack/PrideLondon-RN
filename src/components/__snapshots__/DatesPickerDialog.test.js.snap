// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Dialog
  applyButtonText="Show XX events"
  headerRight={
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Text
        markdown={false}
        style={Object {}}
        type="text"
      >
        Clear
      </Text>
    </TouchableOpacity>
  }
  onApply={[Function]}
  onCancel={[Function]}
  title="Select dates"
  visible={true}
>
  <DatesPicker
    dateRange={null}
    onChange={[Function]}
  />
</Dialog>
`;
